### PR TITLE
[ARM] Use `RegisterClassInfo::getRegPressureSetLimit`

### PR DESCRIPTION
**Pull request by wangpc-pp**
`RegisterClassInfo::getRegPressureSetLimit` is a wrapper of
`TargetRegisterInfo::getRegPressureSetLimit` with some logics to
adjust the limit by removing reserved registers.

It seems that we shouldn't use `TargetRegisterInfo::getRegPressureSetLimit`
directly, just like the comment "This limit must be adjusted
dynamically for reserved registers" said.

Separate from https://github.com/llvm/llvm-project/pull/118787

### CODE REVIEW

File: llvm/lib/Target/ARM/ARMBaseInstrInfo.cpp [hunk 1]
```diff
@@ -6979,7 +6979,7 @@ bool ARMPipelinerLoopInfo::tooMuchRegisterPressure(SwingSchedulerDAG &SSD,
 
   auto &P = RPTracker.getPressure().MaxSetPressure;
   for (unsigned I = 0, E = P.size(); I < E; ++I)
```
Comment (davemgreen) [inline llvm/lib/Target/ARM/ARMBaseInstrInfo.cpp:6981]: Can you add these RegClasses as excluded:
```
    // Exclude some Neon register classes.
    if (I == ARM::DQuad_with_ssub_0 || I == ARM::DTripleSpc_with_ssub_0 ||
        I == ARM::DTriple_with_qsub_0_in_QPR)
      continue;
```

File: llvm/lib/Target/ARM/ARMBaseInstrInfo.cpp [hunk 2]
```diff
@@ -6979,7 +6979,7 @@ bool ARMPipelinerLoopInfo::tooMuchRegisterPressure(SwingSchedulerDAG &SSD,
...
-    if (P[I] > TRI->getRegPressureSetLimit(*MF, I)) {
```
Comment (davemgreen) [inline llvm/lib/Target/ARM/ARMBaseInstrInfo.cpp:6982]: Remove TRI variable now.

**Review by davemgreen — COMMENTED** [review]
Thanks for separating this out. ARM has a lot of nested register classes with overlapping registers, and the register pressure calculations for FP have never been very accurate. I always wanted to exclude certain register classes (like the one causing problems here), but doing that has never led to performance improvements overall. The register pressures looks somewhat like a random number generator to be honest, and the TRI version can be closer to reality in some places. We might have to make something that helps this test case for the time being and looks into the wider problem of fp reg-pressure later on.

File: llvm/lib/Target/ARM/ARMBaseInstrInfo.cpp (hunk 1 above)
Comment (wangpc-pp) [reply llvm/lib/Target/ARM/ARMBaseInstrInfo.cpp]: Thanks, Done!

**Review by davemgreen — APPROVED** [review]
Thanks. LGTM

**Pull request merged**